File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> count{};

        int res{};

        int l{};

        for (int r{}; r < s.length(); r++) {
            count[s[r]]++; 

            while (r-l+1-max_freq(count) > k) {
                count[s[l]]--;
                l++;
            }
            res = max(res, r-l+1);
        }
        return res;
    }

    int max_freq(unordered_map<char, int> m) {
        int max{};

        for (const auto& [key, value]: m) {
            if (value >= max) {
                max = value;
            }
        }
        return max; 
    }
};

File: Data Structures & Algorithms/largest-3-same-digit-number-in-string/submission-0.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        res = -1

        curr = num[0]
        count = 0

        for c in num:
            if c == curr:
                count+=1
                if count == 3:
                    res = max(res, int(c))
            else:
                curr = c
                count = 1
        return str(res)*3 if res != -1 else ""

        